src/main/java/hwr/oop/riddler/model/Sudoku.java

```diff
@@ -1,14 +1,67 @@
 package hwr.oop.riddler.model;
 
 import hwr.oop.riddler.model.component.Box;
+import hwr.oop.riddler.model.component.Cell;
 import hwr.oop.riddler.model.component.Column;
 import hwr.oop.riddler.model.component.Row;
 
+import java.util.ArrayList;
+
 public class Sudoku {
+    public Cell[][] cells = new Cell[9][9];
     public Row[] rows = new Row[9];
     public Column[] columns = new Column[9];
-    public Box[] box = new Box[9];
+    public Box[] boxes = new Box[9];
 
     public Sudoku(int[][] input) {
+        this.fillCells(input);
+        this.interpretRows();
+        this.interpretColumns();
+        this.interpretBoxes();
+    }
+
+    private void fillCells(int[][] input) {
+        for (int rowIndex = 0; rowIndex < input.length; rowIndex++) {
+            int[] row = input[rowIndex];
+            for (int columnIndex = 0; columnIndex < row.length; columnIndex++) {
+                int cellValue = row[columnIndex];
+                this.cells[rowIndex][columnIndex] = new Cell(cellValue);
+            }
+        }
+    }
+
+    private void interpretRows() {
+        for (int rowIndex = 0; rowIndex < this.cells.length; rowIndex++) {
```

Comment:
Hier vielleicht var verwenden statt int, damit man consistent ist

Reply:
var würde ich nur bei Klassennamen verwenden, um den clutter zu vermeiden, wenn mann den Klassennamen doppelt schreibt. Wenn es den code nicht übersichtlicher macht, würde ich lieber (wenn möglich) den konkreten typ nehmen. Soll wohl auch sicherer sein in manchen Situationen.

Reply:
Ja okay auch gutes Argument, ist wahrscheinlich dann Geschmackssache, wir können ja Neubert auch mal fragen, was die da besser finden!